db-service/lib/cqn2sql.js

```diff
@@ -266,32 +258,36 @@ class CQN2SQLRenderer {
    * @returns {string} SQL
    */
   column_expr(x) {
-    if (x.func && !x.as) x.as = x.func
+    if (x === '*') return '*'
+    ///////////////////////////////////////////////////////////////////////////////////////
+    // REVISIT: that should move out of here!
     if (x?.element?.['@cds.extension']) {
-      x.as = x.as || x.element.name
-      return `extensions__->${this.string('$."' + x.element.name + '"')}`
+      return `extensions__->${this.string('$."' + x.element.name + '"')} as ${x.as || x.element.name}`
     }
+    ///////////////////////////////////////////////////////////////////////////////////////
     let sql = this.expr(x)
+    let alias = this.column_alias4(x)
+    if (alias) sql += ' as ' + this.quote(alias)
     return sql
   }
 
+  column_alias4(x) {
```

Comment:
Hmm, can we change the ts settings to not require us to document everything → that is not what we should do, will harm our ability to rafactor

Reply:
The typescript validation step is something we can turn off. It currently only considers APIs exported over the `index.js`, because `CQN2SQL` is part of the `SQLService` it is exported. Which means that all the functions part of the class are exported.

What I found is that when running the validation locally it is very handy to find gaps. So if we turn it off now and turn it on after we have refactored. It will just give a list of functions that require type definitions.

Reply:
I really dislike being forced into the corset of a statically typed language now. We would have chosen typescript if we wanted that. → My proposal:

1. When add jsdoc comments these go into .d.ts
2. When we don't the method is considered private 

Following our rules so far: Only the things we document explicitly are public API.

Reply:
Also needs a call, I think